File: 프로그래머스/0/120868. 삼각형의 완성조건 （2）/삼각형의 완성조건 （2）.java
class Solution {
    public int solution(int[] sides) {
        int ret = 0;
        int big = Math.max(sides[0], sides[1]);
        int small = Math.min(sides[0], sides[1]);
        int sum = sides[0] + sides[1];
        for(int i = 1; i < big; i++) {
            if(i + small > big) ret++;
        }
        for(int i = big; i < sum; i++){
            if(big + small > i) ret++;
        }
        return ret;
    }
}
/*
1. 둘중 큰게 가장 긴변일 때
2. 가장 긴변이 다른 거일 때
3. 둘중 큰것과 길이가 같을 때
*/

File: 프로그래머스/0/120868. 삼각형의 완성조건 （2）/삼각형의 완성조건 （2）.py
def solution(sides):
    big = max(sides)
    small = min(sides)
    
    return sum(1 for i in range(1, big) if (i + small) > big) + sum(1 for i in range(big, small + big) if i < (small +big))
# 두변중 큰놈이 가장 큰놈이여서 가장 큰놈 보다 작은 수중 되는 놈들 숫자 구하기
# 두변의 합보다 작은 놈중 지금 맥스보다 큰놈중 구하기